cpp/examples/tpch/q1.cpp

```diff
@@ -0,0 +1,158 @@
+
+select
+    l_returnflag,
+    l_linestatus,
+    sum(l_quantity) as sum_qty,
+    sum(l_extendedprice) as sum_base_price,
+    sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,
+    sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge,
+    avg(l_quantity) as avg_qty,
+    avg(l_extendedprice) as avg_price,
+    avg(l_discount) as avg_disc,
+    count(*) as count_order
+from
+    lineitem
+where
+    l_shipdate <= date '1998-09-02'
+group by
+    l_returnflag,
+    l_linestatus
+order by
+    l_returnflag,
+    l_linestatus;
+*/
+
+std::unique_ptr<cudf::column> calc_disc_price(std::unique_ptr<table_with_cols>& table) {
+    auto one = cudf::fixed_point_scalar<numeric::decimal64>(1);
+    auto discount = table->column("l_discount");
+    auto one_minus_discount = cudf::binary_operation(one, discount, cudf::binary_operator::SUB, discount.type());
+    auto extended_price = table->column("l_extendedprice");
+    auto disc_price_scale = cudf::binary_operation_fixed_point_scale(
+        cudf::binary_operator::MUL,
+        table->column_type("l_extendedprice").scale(),
+        one_minus_discount->type().scale()
+    );
+    auto disc_price_type = cudf::data_type{cudf::type_id::DECIMAL64, disc_price_scale};
+    auto disc_price = cudf::binary_operation(extended_price, one_minus_discount->view(), cudf::binary_operator::MUL, disc_price_type);
+    return disc_price;
+}
+
+std::unique_ptr<cudf::column> calc_charge(std::unique_ptr<table_with_cols>& table, std::unique_ptr<cudf::column>& disc_price) {
+    auto one = cudf::fixed_point_scalar<numeric::decimal64>(1);
+    auto tax = table->column("l_tax");
+    auto one_plus_tax = cudf::binary_operation(one, tax, cudf::binary_operator::ADD, tax.type());
+    auto charge_scale = cudf::binary_operation_fixed_point_scale(
+        cudf::binary_operator::MUL,
+        disc_price->type().scale(),
+        one_plus_tax->type().scale()
+    );
+    auto charge_type = cudf::data_type{cudf::type_id::DECIMAL64, charge_scale};
+    auto charge = cudf::binary_operation(disc_price->view(), one_plus_tax->view(), cudf::binary_operator::MUL, charge_type);
+    return charge;
+}
+
+int main(int argc, char const** argv) {
+    auto args = parse_args(argc, argv);
+
+    // Use a memory pool
+    auto resource = create_memory_resource(args.memory_resource_type);
+    rmm::mr::set_current_device_resource(resource.get());
+
+    Timer timer;
+
+    // Read out the `lineitem` table from parquet file
+    auto shipdate_ref = cudf::ast::column_reference(5);
+    auto shipdate_upper = cudf::timestamp_scalar<cudf::timestamp_D>(days_since_epoch(1998, 9, 2), true);
+    auto shipdate_upper_literal = cudf::ast::literal(shipdate_upper);
+    auto shipdate_pred = std::make_unique<cudf::ast::operation>(
+        cudf::ast::ast_operator::LESS_EQUAL,
+        shipdate_ref,
+        shipdate_upper_literal
+    );
+    auto lineitem = read_parquet(
```

Comment:
```suggestion
    auto line_item = read_parquet(
```
libcudf follows snake_case naming pattern in this case. See
https://github.com/rapidsai/cudf/blob/branch-24.08/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#code-and-documentation-style-and-formatting

Reply:
I was actually trying to match the table names in TPC-H and it's called `lineitem` as a whole in there.